src/libcore/mem.rs

```diff
@@ -177,15 +177,57 @@ pub fn forget<T>(t: T) {
 
 /// Returns the size of a type in bytes.
 ///
-/// More specifically, this is the offset in bytes between successive
-/// items of the same type, including alignment padding.
+/// More specifically, this is the offset in bytes between successive elements
+/// in an array with that item type including alignment padding. Thus, for any
+/// type `T` and length `n`, [T; n] will have a size of `n * size_of<T>()`. 
+///
+/// In general, the size of a type is not stable across compilations, but
+/// specific types such as primitives are.
+///
+/// The following table gives the size for primitives.
+///
+/// Type | size_of<Type>()
+/// ---- | ---------------
+/// () | 0
+/// u8 | 1
+/// u16 | 2
+/// u32 | 4
+/// u64 | 8
+/// i8 | 1
+/// i16 | 2
+/// i32 | 4
+/// i64 | 8
+/// f32 | 4
+/// f64 | 8
+/// char | 4
+///
+/// Furthermore, `usize` and `isize` will have the same size.
+///
+/// The types `*T`, `&T`, and `Box<T>` have the same size
+/// and `Option`s of `&T` and `Box<T>` have the same size as the pointer.
```

Comment:
I found this a little hard to parse. A skeptical reader would read this as:

```
Options    &T
       \  /
        of     Box<T>
          \   /
           and                       the pointer
              \                     /
               have the same size as
```

instead of

```
        &T     Box<T>
          \   /
Options    and
       \  /
        of                       the pointer
          \                     /
           have the same size as
```

because that sounds more plausible to someone skeptical of zero overhead abstractions.

Is there a way to word this that shows the full types `Option<&T>` and `Option<Box<T>>` in the sentence?

Maybe something straightforward like:

> The types *T, &T, Box\<T>, Option\<&T>, and Option\<Box\<T>> all have the same size. If T is [Sized](https://doc.rust-lang.org/std/marker/trait.Sized.html), all of those have the same size as usize.

Reply:
Yeah, I was sort of floundering with that section. That reads a lot better.